core/include/moveit/task_constructor/stages/deep_grasp_pose.h

```diff
@@ -0,0 +1,299 @@
+/*********************************************************************
+ * BSD 3-Clause License
+ *
+ * Copyright (c) 2020 PickNik Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Boston Cleek
+   Desc:   Grasp generator stage using deep learning based grasp synthesizers
+*/
+
+#pragma once
+
+#include <moveit/task_constructor/stages/generate_pose.h>
+#include <moveit/task_constructor/stages/action_base.h>
+#include <moveit/task_constructor/storage.h>
+#include <moveit/task_constructor/marker_tools.h>
+#include <rviz_marker_tools/marker_creation.h>
+#include <moveit/planning_scene/planning_scene.h>
+
+// #include <memory>
+#include <functional>
+// #include <iostream>
+
+namespace moveit {
+namespace task_constructor {
+namespace stages {
+
+constexpr char LOGNAME[] = "deep_grasp_generator";
+
+/**
+ * @brief Generate grasp candidates using deep learning approaches
+ * @param ActionSpec - action message (action message name + "ACTION")
+ * @details Interfaces with a deep learning based grasp library using a action client
+ */
+template <class ActionSpec>
+class DeepGraspPose : public GeneratePose, ActionBase<ActionSpec>
+{
+private:
+	typedef ActionBase<ActionSpec> ActionBaseT;
+	ACTION_DEFINITION(ActionSpec);
```

Comment:
Isn't this redundant, because the action types are pulled in from `ActionBase<ActionSpec>`?